src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -539,6 +511,50 @@ private static void ExecuteAndCompleteQuery(QueryExecuteParams executeParams, Re
             }
         }
 
+        // Internal for testing purposes
+        internal string GetSqlText(ExecuteRequestParamsBase request)
+        {
+            // If it is a document selection, we'll retrieve the text from the document
+            ExecuteDocumentSelectionParams docRequest = request as ExecuteDocumentSelectionParams;
+            if (docRequest != null)
+            {
+                // Get the document from the parameters
+                ExecuteDocumentSelectionParams documentRequest = (ExecuteDocumentSelectionParams) request;
```

Comment:
I can see where you're coming from, but I think the logic is small enough that it doesn't need to be broken out into a separate set of methods. If we ever get more types to check against, I'd be more apt to consider it, but for now I think it's fine as-is.

Definitely agree on the redundant declarations. Coffee hadn't kicked in this morning. Will address right away.